api/src/modules/custom-projects/input-factory/custom-project-input.factory.ts

```diff
@@ -0,0 +1,168 @@
+} from '@api/modules/custom-projects/dto/conservation-project-params.dto';
+import { CarbonInputs } from '@api/modules/calculations/data.repository';
+import { LOSS_RATE_USED } from '@shared/schemas/custom-projects/create-custom-project.schema';
+import {
+  CARBON_REVENUES_TO_COVER,
+  CreateCustomProjectDto,
+} from '@api/modules/custom-projects/dto/create-custom-project-dto';
+import { ECOSYSTEM } from '@shared/entities/ecosystem.enum';
+import { CostInputs } from '@api/modules/custom-projects/dto/project-cost-inputs.dto';
+import { CustomProjectAssumptionsDto } from '@api/modules/custom-projects/dto/project-assumptions.dto';
+
+export type ConservationProjectCarbonInputs = {
+  lossRate: number;
+  emissionFactor: number | null;
+  emissionFactorAgb: number | null;
+  emissionFactorSoc: number | null;
+};
+
+export type GeneralProjectInputs = {
+  projectName: CreateCustomProjectDto['projectName'];
+  countryCode: CreateCustomProjectDto['countryCode'];
+  activity: CreateCustomProjectDto['activity'];
+  ecosystem: CreateCustomProjectDto['ecosystem'];
+  projectSizeHa: CreateCustomProjectDto['projectSizeHa'];
+  initialCarbonPriceAssumption: CreateCustomProjectDto['initialCarbonPriceAssumption'];
+  carbonRevenuesToCover: CreateCustomProjectDto['carbonRevenuesToCover'];
+};
+
+@Injectable()
+export class CustomProjectInputFactory {
+  createProject(dto: CreateCustomProjectDto, carbonInputs: CarbonInputs) {
+    if (dto.activity === ACTIVITY.CONSERVATION) {
+      //return new ConservationProject(projectConfig);
+      return this.createConservationProjectInput(dto, carbonInputs);
+    } else if (dto.activity === ACTIVITY.RESTORATION) {
+      throw new NotImplementedException('Restoration not implemented');
+    } else {
+      throw new Error('Invalid activity type');
+    }
+  }
+
+  private createConservationProjectInput(
+    dto: CreateCustomProjectDto,
+    carbonInputs: CarbonInputs,
+  ): ConservationProjectInput {
+    const {
+      parameters,
+      assumptions,
+      costInputs,
+      projectName,
+      projectSizeHa,
+      initialCarbonPriceAssumption,
+      activity,
+      carbonRevenuesToCover,
+      ecosystem,
+      countryCode,
+    } = dto;
+
+    const conservationProjectInput: ConservationProjectInput =
+      new ConservationProjectInput();
+    conservationProjectInput.setGeneralInputs({
+      projectName,
+      projectSizeHa,
+      initialCarbonPriceAssumption,
+      activity,
+      carbonRevenuesToCover,
+      ecosystem,
+      countryCode,
+    });
+    conservationProjectInput.setLossRate(
+      parameters as ConservationProjectParamDto,
+      carbonInputs,
+    );
+    conservationProjectInput.setEmissionFactor(
+      parameters as ConservationProjectParamDto,
+      carbonInputs,
+    );
+    conservationProjectInput.setCostInputs(costInputs);
+    conservationProjectInput.setModelAssumptions(assumptions);
+
+    return conservationProjectInput;
+  }
+}
+
+export class ConservationProjectInput {
```

Comment:
We should have this class somewhere else, but I'm not sure if this approach is too convoluted and we should simplify the building of the input more, maybe using some well structured pattern?

Any suggestions? We are tight on time but I think it's important to make the actual implementation right so we can modify it later on